Recent changes don't compile with MSVC, cause warnings in GCC

The additions in #61 fail to compile with MSVC. Is MSVC supported?

```
# using VS2022 (17.7.34031.279) ...
deps\simdsort\src\xss-network-qsort.hpp(80,11): error C2466: cannot allocate an array of constant size 0
simdsort\src\xss-network-qsort.hpp(76,9): message : see reference to function template instantiation 'void sort_n_vec<vtype,0,zmm_vector<float>::reg_t>(zmm_vector<float>::type_t *,int32_t)' b 
eing compiled
          with
          [
              vtype=vtype
          ]
simdsort\src\xss-network-qsort.hpp(76,9): message : see reference to function template instantiation 'void sort_n_vec<vtype,1,zmm_vector<float>::reg_t>(zmm_vector<float>::type_t *,int32_t)' b 
eing compiled
          with
          [
              vtype=vtype
          ]
simdsort\src\xss-network-qsort.hpp(76,9): message : see reference to function template instantiation 'void sort_n_vec<vtype,2,zmm_vector<float>::reg_t>(zmm_vector<float>::type_t *,int32_t)' b 
eing compiled
          with
          [
              vtype=vtype
          ]
simdsort\src\xss-network-qsort.hpp(76,9): message : see reference to function template instantiation 'void sort_n_vec<vtype,4,zmm_vector<float>::reg_t>(zmm_vector<float>::type_t *,int32_t)' b 
eing compiled
          with
          [
              vtype=vtype
          ]
simdsort\src\xss-network-qsort.hpp(76,9): message : see reference to function template instantiation 'void sort_n_vec<vtype,8,zmm_vector<float>::reg_t>(zmm_vector<float>::type_t *,int32_t)' b 
eing compiled
          with
          [
              vtype=vtype
          ]
simdsort\src\xss-network-qsort.hpp(134,5): message : see reference to function template instantiation 'void sort_n_vec<vtype,16,zmm_vector<float>::reg_t>(zmm_vector<float>::type_t *,int32_t)' 
 being compiled
          with
          [
              vtype=vtype
          ]
xxx.h(199,5): message : see reference to function template instantiation 'void sort_n<vtype,256>(zmm_vector<float>::type_t *,int)' being compiled
simdsort\src\xss-network-qsort.hpp(80,11): error C2133: 'vecs': unknown size
simdsort\src\xss-network-qsort.hpp(83,30): error C2466: cannot allocate an array of constant size 0
simdsort\src\xss-network-qsort.hpp(83,30): error C2133: 'ioMasks': unknown size
```

The changes also cause a few GCC `-Wunused-but-set-parameter` warnings for the `regs` parameter in the same file.

> The changes also cause a few GCC `-Wunused-but-set-parameter` warnings for the `regs` parameter in the same file.

These appear to be resolved by using gcc-10/g++-10 instead of 9.

```
../deps/simdsort/src/xss-network-qsort.hpp: In instantiation of ‘void bitonic_fullmerge_n_vec(reg_t*) [with vtype = zmm_vector<float>; long int numVecs = 16; long int numPer = 32; reg_t = __vector(16) float]’:
../deps/simdsort/src/xss-network-qsort.hpp:68:60:   recursively required from ‘void bitonic_fullmerge_n_vec(reg_t*) [with vtype = zmm_vector<float>; long int numVecs = 16; long int numPer = 4; reg_t = __vector(16) float]’
../deps/simdsort/src/xss-network-qsort.hpp:68:60:   required from ‘void bitonic_fullmerge_n_vec(reg_t*) [with vtype = zmm_vector<float>; long int numVecs = 16; long int numPer = 2; reg_t = __vector(16) float]’
../deps/simdsort/src/xss-network-qsort.hpp:113:44:   required from ‘void sort_n_vec(typename vtype::type_t*, int32_t) [with vtype = zmm_vector<float>; int numVecs = 16; reg_t = __vector(16) float; typename vtype::type_t = float; int32_t = int]’
../deps/simdsort/src/xss-network-qsort.hpp:136:31:   required from ‘void sort_n(typename vtype::type_t*, int) [with vtype = zmm_vector<float>; long int maxN = 256; typename vtype::type_t = float]’
../src/etc/gm.h:200:55:   required from here
../deps/simdsort/src/xss-network-qsort.hpp:59:58: warning: parameter ‘regs’ set but not used [-Wunused-but-set-parameter]
```

--- a/src/xss-network-qsort.hpp
+++ b/src/xss-network-qsort.hpp
@@ -72,9 +72,11 @@ X86_SIMD_SORT_UNROLL_LOOP(64)
 template <typename vtype, int numVecs, typename reg_t = typename vtype::reg_t>
 X86_SIMD_SORT_INLINE void sort_n_vec(typename vtype::type_t *arr, int32_t N)
 {
-    if (numVecs > 1 && N * 2 <= numVecs * vtype::numlanes) {
-        sort_n_vec<vtype, numVecs / 2>(arr, N);
-        return;
+    if constexpr (numVecs > 1) {
+        if (N * 2 <= numVecs * vtype::numlanes) {
+            sort_n_vec<vtype, numVecs / 2>(arr, N);
+            return;
+        }
     }
 
     reg_t vecs[numVecs];

